### PR TITLE
chore(.eslintignore): add eslint ignoring rule

### PR DESCRIPTION
**Pull request by manudeli**
I added eslint ignoring rule like [nextjs](https://github.com/vercel/next.js/blob/canary/.eslintignore)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+build
```
Comment (TkDodo) [inline .eslintignore:2]: we don't have a top level build directory, and `node_modules` is always excluded. I don't think this does anything

Comment (manudeli) [reply .eslintignore]: Oh, I didn't think about top level. I excluded top level `build` directory in `.eslintignore`. and remove `node_modules`

File: .gitignore [hunk 2]
```diff
@@ -39,9 +38,7 @@ stats.html
 
 *.log
 .DS_Store
-node_modules
```
Comment (TkDodo) [inline .gitignore:42]: can you please leave node_modules in gitignore

Comment (manudeli) [reply .gitignore:42]: Ah, I should have left a comment explaining the change. You can check duplicate node_modules in `.gitignore`. so I removed it. https://github.com/TanStack/query/pull/5813/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947L5

removing `dist` have same reason

**Review by TkDodo — APPROVED** [review]

**Pull request merged**